js/plates-model/volcanic-activity.ts

```diff
@@ -171,29 +162,30 @@ export default class VolcanicActivity {
         if (blob.active) {
           blob.active = false;
         }
-        if (blob.canErupt) {
-          if (this.intensity > 0 && this.eruptionTime === 0 && this.eruptionCooldown === 0) {
+        const canErupt = blob.maxDist === crustThickness; // magma can erupt when it reaches the surface
+        if (canErupt) {
+          if (this.intensity > 0 && this.eruptionTime === 0) {
+            blob.isErupting = true;
             this.eruptionTime = ERUPTION_TIME;
           }
-          if (this.eruptionTime > 0) {
-            // Sometimes it's necessary to move field up to line it up with the surface.
-            blob.dist = crustThickness;
-          }
+        }
+        if (blob.isErupting) {
+          // Line up erupting magma blob with the crust surface.
+          blob.dist = crustThickness;
         }
       }
     });
 
-    if (this.eruptionCooldown > 0) {
-      this.eruptionCooldown = Math.max(0, this.eruptionCooldown - timestep);
-    }
-
     if (this.eruptionTime > 0) {
       this.eruptionTime -= timestep;
-
-      if (this.eruptionTime <= 0) {
-        this.eruptionTime = 0;
-        this.eruptionCooldown = ERUPTION_COOLDOWN;
-      }
+    }
+    if (this.eruptionTime <= 0 || !this.active) {
+      this.eruptionTime = 0;
+      this.magma.forEach(blob => {
+        if (blob.isErupting) {
+          blob.isErupting = false;
+        }
```

Comment:
Is this functionally different than the simpler:
```suggestion
        blob.isErupting = false;
```
I thought MobX was smart enough to not notify observers if the value didn't actually change. Anyway, more a question than a recommendation.

Reply:
Actually, there's no MobX anywhere in the plates-model dir. It's all model files used by the web worker. I guess I've added this `if` to limit the number of writes (as theoretically should be slower than value comparison 😉).  Usually, this would deserve to be called be premature optimization, but maybe it won't hurt in files in `plates-model` dir.